lib/Screencast.js

```diff
@@ -0,0 +1,236 @@
+/**
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const fs = require('fs');
+const EventEmitter = require('events');
+const {helper, debugError} = require('./helper');
+
+/**
+ * @param {!Array<number>} frames
+ * @param {number} fps Desired frames per second.
+ * @return {!Promise<Blob>}
+ */
+async function createVideo(frames, fps) {
+  const imageBitmaps = await Promise.all(frames.map(frame => {
+    const blob = new Blob([Uint8Array.from(frame)], {type: 'image/png'});
+    return window.createImageBitmap(blob);
+  }));
+
+  // Use last image as width so we know the set viewport is correct.
+  const WIDTH = imageBitmaps[imageBitmaps.length - 1].width;
+  const HEIGHT = imageBitmaps[imageBitmaps.length - 1].height;
+
+  const canvas = document.createElement('canvas');
+  canvas.style.display = 'none';
+  canvas.width = WIDTH;
+  canvas.height = HEIGHT;
+  canvas.style.width = `${WIDTH / 2}px`;
```

Comment:
Typically you scale a canvas according to the DPR of the device so the images are clear. IOW, if the screencast api captured images at 1000x800 and DPR was 2, you'd want to create a 500x400 canvas. I think we can probably remove this since 1.) we're not accounting for DPR here and 2.) The DTP screen API doesn't respect DPR.